crates/wascc-provider/src/states.rs

```diff
@@ -8,3 +8,28 @@ pub(crate) mod running;
 pub(crate) mod starting;
 pub(crate) mod terminated;
 pub(crate) mod volume_mount;
+
+/// When called in a state's `next` function, exits the current state
+/// and transitions to the Error state.
+#[macro_export]
+macro_rules! transition_to_error {
```

Comment:
This looks pretty good, maybe a shorter name. 
Could we also introduce something to unwrap `Option` and `Result` easily?

Reply:
https://github.com/kflansburg/krustlet/commit/b598697c31c5c1628df79372c0212171d20889d6

Reply:
Passing in `self` is becoming very tiresome :(

Reply:
I started with macros that wrapped `Result` so you could write `let handle = fatal_if_failed!(start_container(...))` but I didn't like it because it buried the lede especially in the error case where you had to write `let foo = error_if_failed!(self, some_complex_expression(...))` and the _thing you were actually doing on the line_ got buried in noise.

I looked at the Try trait so that we could have the modifier in suffix position but that didn't work...